Model/Indexer/Category/ProductIndexer.php

```diff
@@ -144,6 +151,13 @@ public function executeByDimensions(array $dimensions, \Traversable $entityIds):
         }
 
         $entityIds = iterator_to_array($entityIds);
+        // Include configurable product id(s) if the edited product is simple
+        foreach ($entityIds as $entityId) {
+            $parentEntityIds = $this->productIndexerHelper->getParentEntityId($entityId);
+            if (!empty($parentEntityIds)) {
+                $entityIds = array_merge($entityIds, $parentEntityIds);
```

Comment:
As mentioned on the other class, you can actually pass the whole array of entity ids to the `Configurable` class, and get back the set of parent ids to be added.
If, however, we could only do it one at a time, we want to avoid calling `array_merge` inside a `foreach`.

What you can do instead (for future reference) is:
```
$parentEntityIds = [];
foreach ($entityIds as $entityId) {
    $parentEntityIds[] = $this->productIndexerHelper->getParentEntityId($entityId);
}
$entity_ids = array_merge($entityIds, ...$parentEntityIds);
```